bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonEncoder.kt

```diff
@@ -156,6 +156,8 @@ internal class DefaultBsonEncoder(
             if (value != null || configuration.explicitNulls) {
                 encodeName(it)
                 super.encodeNullableSerializableValue(serializer, value)
+            } else {
```

Comment:
It seems worth it to add tests for the corresponding `else` clause in the `encodeSerializableValue` method just above (I removed the clause and no tests fails, so it's not covered).

Reply:
For me the new test: `KotlinSerializerCodecTest.testDataClassWithListThatLastItemDefaultsToNull` fails if I remove this else statement.

Reply:
nm - will update the tests.

Reply:
I can't concoct a test that triggers this.  That said it doesn't guarantee that there isn't a scenario where it could be triggered. 

As we are skipping this value it makes sense to reset the defferred name if it exists.
However, I think I can clean up the usage.

Reply:
@jyemin I cleaned up the usage and encapsulated the handling of deferred element names.